mmv1/third_party/terraform/resources/resource_container_cluster.go.erb

```diff
@@ -78,6 +78,13 @@ var (
 	forceNewClusterNodeConfigFields = []string{
 		"workload_metadata_config",
 	}
+
+	suppressDiffForAutopilot = schema.SchemaDiffSuppressFunc(func(k, oldValue, newValue string, d *schema.ResourceData) bool {
```

Comment:
Just want to confirm if my understanding is correct. This DSF is added to prevent premadiff for some fields that return computed values when `enable_autopilot == true`. If so, do you think making the field optional + computed will solve the problem?

Reply:
The problem is that these fields have default values, which conflicts with optional/computed. We can't remove the default values, but they will be ignored for Autopilot and the result is a permadiff

Reply:
Ah, like for autopilot, the API will not return those values?

Reply:
It returns different values, or no values for different fields. The intent here is essentially the field is optional + computed only when `enable_autopilot == true`

Reply:
Got it! Thanks! Then it sounds like that optional + computed make more sense for such fields than having default values on Terraform side. 
For cases other than `enable_autopilot == true`, do these fields have the same server side defaults as the Terraform side ones. If so, I guess it would be relatively safe to change default values to optional + computed.

Reply:
We don't have the same defaults when Autopilot is enabled, and these may change over time as these values are implementation details for Autopilot, whereas they matter directly in GKE Standard.

This is another side effect of the dual purpose cluster APIs, but when Autopilot is enabled we cannot make any assumptions about server side values as their future changes are _not_ considered breaking

Reply:
Discussed offline - The fields are required when Autopilot is not enabled. So changing to optional + computed will not work. DSF may be the best way to handle perma-diffs in this case.